### DISCUSSION TITLE
Warnings vs Spikes ?

### DISCUSSION THREAD

**Discussion by GrahamLea** [amberelectric/public-api#110] [Q&A]
Hiya. I'm building a new dashboard webapp and noticing a discrepancy between my output and `app.amber.com.au`.

I'm seeing warning icons in app.amber.com.au:

<img width="697" alt="image" src="https://user-images.githubusercontent.com/754403/166927273-0e406354-4294-4a23-91dd-29adc894ee07.png">

but in my API data I don't see a `spike` descriptor or a positive `spikeStatus`:
```
{
    "type": "ForecastInterval",
    "date": "2022-05-06",
    "duration": 30,
    "startTime": "2022-05-05T20:00:01Z",
    "endTime": "2022-05-05T20:30:00Z",
    "nemTime": "2022-05-06T06:30:00+10:00",
    "perKwh": 45.76202,
    "renewables": 2.801374813043719,
    "spotPerKwh": 32.9989,
    "channelType": "general",
    "spikeStatus": "none",
    "tariffInformation": {
        "period": "offPeak"
    },
    "descriptor": "high",
    "range": {
        "min": 29.999,
        "max": 1509.998
    }
}
```

So is the warning icon based off the `range.max`?
If I wanted to replicate the logic used in the app for the warning, what is it?
I'm guessing something like `interval.spikeStatus || interval.range.max > 300` ?

**Comment by madpilot** [comment]
The interval you are showing in the API is 6:00am-6:30am, which isn't a spike. (The time in the web UI is the start of the period, NEM time is the end)

Though, the price does seem out (you are in NSW, right?) - it should show 71c for it to match.

I'll check it in the morning.

NEM time 7:00 and 7:30 should show spike status.

**Comment by GrahamLea** [comment]
Yeah, I definitely forgot NEM Time was the _end_ time. 🤦🏻‍♂️

I could have sworn though that I was seeing warning icons on your website, but no spikes in the API data. But, I've been copying data from the API intermittently while prototyping, so it's possible they got out of sync.

Yes, I'm in NSW.
I read in another thread that the spike status is based on the spot price predication, which is not high in the above data, despite the potentially high max range, so I'm guessing that's why it doesn't predict a spike.

**Reply by madpilot** [reply]
NEM time is one of the biggest sources of bugs for us. It still trips me up occasionally.

For forecasts, the spike attribute will never be "potential", it'll be "spike" of the spotPrice is greater than $3, otherwise "none".

Current Interval is a little more complicated: it'll be "potential" if the 30 minute average is a spike, but no 5 minutes intervals have actually spiked yet. One a 5 minute interval does spike, it gets upgraded to "spike".

**Comment by GrahamLea** [comment]
Thanks for that, Myles.

For the warning/spike/flashing level on my dashboard, I ended up checking if the spike status is "spike" or "potential", if the descriptor is "spike", or if the perKwh is above a user-defined threshold, defaulted to $1.

Graham Lea

Co-Founder

+61 404 147 684

***@***.*** ( ***@***.*** )

https://archium.io/ ( https://archium.io/ )

Sent via Superhuman ( ***@***.*** )
